electrumsv/app_state.py

```diff
@@ -91,8 +90,21 @@ def set_app(self, app) -> None:
     def headers_filename(self) -> str:
         return os.path.join(self.config.path, 'headers')
 
+    def setup_regtest(self) -> HeadersRegTestMod:
+        regtest_import_privkey_to_node()
+        delete_headers_file(self.headers_filename())
+        Net.CHECKPOINT, Net.VERIFICATION_BLOCK_MERKLE_ROOT = calculate_regtest_checkpoint(
+            Net.MIN_CHECKPOINT_HEIGHT)
+        logger.info("using regtest network - miner funds go to: '%s' (not part of this wallet)",
+                    Net.REGTEST_P2PKH_ADDRESS)
+        logger.info("top-up via 'regtest_topup_account' method")
+        return HeadersRegTestMod.from_file(Net.COIN, self.headers_filename(), Net.CHECKPOINT)
+
     def read_headers(self) -> None:
-        self.headers = Headers.from_file(Net.COIN, self.headers_filename(), Net.CHECKPOINT)
+        if self.config.get('regtest'):
```

Comment:
If it is possible to tell by looking what network we are using by looking at it, we should do so rather than looking at config I think.

Reply:
Any suggestions? This fails - don't know why.
```
>>> isinstance(Net._net, SVRegTestnet)
False
```

Reply:
Other than this one, all other suggestions are committed